src/csvcubed/cli/inspect/metadataprocessor.py

```diff
@@ -99,14 +121,110 @@ def load_json_ld_to_rdflib_graph(self) -> Graph:
+            parse_graph_retain_relative(
                 data=csvw_file_content,
-                publicID=path_to_file_uri_for_rdflib(csvw_metadata_file_path),
                 format="json-ld",
+                graph=csvw_metadata_rdf_graph.get_context(
+                    path_to_file_uri_for_rdflib(self.csvw_metadata_file_path)
+                ),
             )
+
             _logger.info("Successfully parsed csvw json-ld to rdf graph.")
 
-            self._load_table_schema_dependencies_into_rdf_graph(csvw_metadata_rdf_graph)
+            self._load_table_schema_dependencies_into_rdf_graph(
+                csvw_metadata_rdf_graph, csvw_metadata_file_path
+            )
+
+            add_triples_for_file_dependencies(
+                csvw_metadata_rdf_graph, self.csvw_metadata_file_path
+            )
+
             return csvw_metadata_rdf_graph
         except Exception as ex:
             raise FailedToLoadRDFGraphException(self.csvw_metadata_file_path) from ex
+
+
+def add_triples_for_file_dependencies(
+    rdf_graph: rdflib.ConjunctiveGraph,
+    paths_relative_to: Union[str, Path],
+    follow_relative_path_dependencies_only: bool = False,
+) -> None:
+    """
+    Loads dependent RDF metadata files, along with transitive dependencies.
+
+    This is exposed publicly for re-use by the csvcubed-pmd project.
+    """
+
+    _logger.debug("Loading RDF dependencies")
+
+    dependencies_to_load = select_metadata_dependencies(rdf_graph)
+
+    paths_relative_to_str = (
+        path_to_file_uri_for_rdflib(paths_relative_to)
+        if isinstance(paths_relative_to, Path)
+        else paths_relative_to
+    )
+
+    if follow_relative_path_dependencies_only and any(dependencies_to_load):
+        _logger.debug("Dropping non-relative dependencies.")
+        dependencies_to_load = [
+            d for d in dependencies_to_load if not looks_like_uri(d.data_dump)
+        ]
+
+    for d in dependencies_to_load:
+        if not looks_like_uri(d.data_dump):
+            absolute_url = urljoin(paths_relative_to_str, d.data_dump)
+            _logger.debug(
+                "Treating relative dependency '%s' as absolute URL '%s'.",
+                d.data_dump,
+                absolute_url,
+            )
+            d.data_dump = absolute_url
+
+    if not any(dependencies_to_load):
+        _logger.debug("Did not find any RDF dependencies to load.")
+
+    for dependency in dependencies_to_load:
+        this_dependency_rdf = rdf_graph.get_context(dependency.data_dump)
+        if any(this_dependency_rdf):
+            _logger.debug(
+                "Skipping dependency '%s' as it has already been loaded.",
+                dependency.data_dump,
+            )
+            continue
+
+        _logger.debug(
+            "Loading dataset dependency '%s' covering uriSpace '%s' in dataset '%s'",
+            dependency.data_dump,
+            dependency.uri_space,
+            dependency.data_set,
```

Comment:
Do we need to have 2 loops here? Can't we do the functionality in the 2nd loop within the 1st loop?

Reply:
Fair point, I've merged the two of them together now.